R/simulation.R

```diff
@@ -290,9 +290,27 @@ setMethod(
     # source module metadata and code files, checking version info
     lapply(modules(sim), function(m) {
       md <- moduleMetadata(m, modulePath(sim))
-      if (md$version != packageVersion("SpaDES")) {
+      mVersion <- unlist(md$version)
+      pVersion <- unlist(packageVersion("SpaDES"))
+      maxLength <- max(length(mVersion), length(pVersion))
+      if(length(mVersion) != length(pVersion)){
+        if(length(mVersion) != maxLength){
+          mVersion[(length(mVersion)+1):maxLength] <- 0
+        } else {
+          pVersion[(length(pVersion)+1):maxLength] <- 0
+        }
+      }
+      maxDigits <- max(max(ceiling(log10(mVersion+1)), na.rm = TRUE),
+                       max(ceiling(log10(pVersion+1)), na.rm = TRUE))
+      mVersionNumeric <- mVersion[maxLength]
+      pVersionNumeric <- pVersion[maxLength]
+      for(i in 1:(maxLength-1)){
+        mVersionNumeric <- mVersionNumeric + mVersion[i]*10^((maxLength-i)*maxDigits)
+        pVersionNumeric <- pVersionNumeric + pVersion[i]*10^((maxLength-i)*maxDigits)
+      }
+      if (mVersionNumeric %>>% pVersionNumeric) {
```

Comment:
Hi Yong, I don't understand -- the `numeric_version` and `package_version` are built in and work as expected.

```r
v1 <- package_version("1.1.0")
v2 <- package_version("1.1.0.9000")
v3 <- numeric_version("1.0.1")
v4 <- numeric_version("1.0.1.9000")

## all work correctly
> (v1 == v2)
[1] FALSE
> (v1 > v2)
[1] FALSE
> (v1 < v2)
[1] TRUE
> 
> (v1 == v3)
[1] FALSE
> (v1 > v3)
[1] TRUE
> (v1 < v3)
[1] FALSE
> 
> (v1 == v4)
[1] FALSE
> (v1 > v4)
[1] TRUE
> (v1 < v4)
[1] FALSE

> (v4 == v1)
[1] FALSE
> (v4 > v1)
[1] FALSE
> (v4 < v1)
[1] TRUE
> 
> (v4 == v2)
[1] FALSE
> (v4 > v2)
[1] FALSE
> (v4 < v2)
[1] TRUE
> 
> (v4 == v3)
[1] FALSE
> (v4 > v3)
[1] TRUE
> (v4 < v3)
[1] FALSE
```